utils/internal/download_gigaspeech_from_speechocean.sh

```diff
@@ -0,0 +1,191 @@
+. ./env_vars.sh || exit 1
+
+# Check credential
+if [ ! -f SAFEBOX/password ]; then
+  echo "$0: Please apply for the download credentials (see the \"Download\""
+  echo "$0: section in README) and it to SAFEBOX/password."
+  exit 1
+fi
+
+PASSWORD=`cat SAFEBOX/password 2>/dev/null`
+if [ -z "$PASSWORD" ]; then
+  echo "$0: Error, SAFEBOX/password is empty."
+  exit 1
+fi
+
+# Check downloading tools
+if ! which wget >/dev/null; then
+  echo "$0: Error, please make sure you have wget installed."
+  exit 1
+fi
+if ! which openssl >/dev/null; then
+  echo "$0: Error, please make sure you have openssl installed."
+  exit 1
+fi
+
+openssl_distro=`openssl version | awk '{print $1}'`
+required_distro="OpenSSL"
+if [[ "$openssl_distro" != "$required_distro" ]]; then
+  echo "$0: Unsupported $openssl_distro detected, please use $required_distro"
+  echo "$0: On mac, you should try: brew install openssl"
+  exit 1
+fi
+
+openssl_version=`openssl version|sed -E 's/^.*([0-9]+\.[0-9]+\.[0-9]+).*$/\1/g'`
+required_version="1.1.1"
+older_version=$(printf "$required_version\n$openssl_version\n"|sort -V|head -n1)
+if [[ "$older_version" != "$required_version" ]]; then
+  echo "$0: The script requires openssl version $required_version or newer."
+  echo "$0: Please download the openssl source code, and install openssl"
+  echo "$0: version $required_version"
+  exit 1
+fi
+
+download_object_from_release() {
+  local obj=$1
+  echo "$0: Downloading $obj"
+  local remote_obj=$obj
+  local location=${gigaspeech_dataset_dir}$obj
+  local location_dirname=$(dirname ${gigaspeech_dataset_dir}/$obj)
+
+  mkdir -p $location_dirname || exit 1;
+
+  # -T seconds timeout, -t number of tries
+  wget -c -t 20 -T 90 --ftp-user=GigaSpeech --ftp-password=$PASSWORD ftp://124.207.81.184/GigaSpeech/$remote_obj -O $location || exit 1;
+}
+
+process_downloaded_object() {
+  local obj=$1
+  echo "$0: Processing $obj"
+  local path=${gigaspeech_dataset_dir}/$obj
+  local location=$(dirname $path)
+
+  if [[ $path == *.tgz.aes ]]; then
+    # encrypted-gziped-tarball contains contents of a GigaSpeech sub-directory
+    local subdir_name=$(basename $path .tgz.aes)
+    mkdir -p $location/$subdir_name || exit 1;
+    openssl aes-256-cbc -d -salt -pass pass:$PASSWORD -pbkdf2 -in $path |\
+      tar xzf - -C $location/$subdir_name || exit 1;
+  elif [[ $path == *.gz.aes ]]; then
+    # encripted-gziped object represents a regular GigaSpeech file
+    local file_name=$(basename $path .gz.aes)
+    mkdir -p $location || exit 1;
+    openssl aes-256-cbc -d -salt -pass pass:$PASSWORD -pbkdf2 -in $path |\
+      gunzip -c > $location/$file_name || exit 1;
+  else
+    # keep the object as it is
+    :
+  fi
+}
+
+
+# User agreement
+if [ $stage -le 0 ]; then
+  echo "$0: Start to download GigaSpeech user agreement"
+  wget -c -t 20 -T 90 --ftp-user=GigaSpeech --ftp-password=$PASSWORD \
+    ftp://124.207.81.184/GigaSpeech/TERMS_OF_ACCESS -O $gigaspeech_dataset_dir/TERMS_OF_ACCESS 
```

Comment:
It feels unsafe to have these IP/FTP address hard-coded into downloading scripts, especially when they appear more than once in the code.  It may cause problems in the future when we want to find/modify them due to public IP changes.  It's better we try to read these concrete IPs from a configurable & centralized resource.